magma/conversions.py

```diff
@@ -170,7 +170,13 @@ def convertbits(value, n, totype, checkbit):
             Direction.InOut: InOut
         }[T.direction](Bit)
 
-    return totype[len(Ts), T](ts)
+    value = totype[len(Ts), T](ts)
+    if n is not None and len(value) < n:
+        if isinstance(value, SInt):
```

Comment:
do these types have `ext_to`? then we could just use that and it'll dispatch to sext/zext? not sure.

Reply:
We could add that, but I originally only added the function style interface